tests/sse/test_sse2.cpp

```diff
@@ -0,0 +1,342 @@
+	Ret_M128d(int64_t, _mm_cvtsd_si64); // _mm_cvtsd_si64x is an alias to this.
+	Ret_M128i(int, _mm_cvtsi128_si32);
+	Ret_M128i(int64_t, _mm_cvtsi128_si64); // _mm_cvtsi128_si64x is an alias to this.
+	Ret_M128d_int(__m128d, _mm_cvtsi32_sd);
+	Ret_int(__m128i, _mm_cvtsi32_si128);
+	Ret_M128d_int64(__m128d, _mm_cvtsi64_sd); // _mm_cvtsi64x_sd is an alias to this.
+	Ret_int64(__m128i, _mm_cvtsi64_si128); // _mm_cvtsi64x_si128 is an alias to this.
+	Ret_M128d_M128(__m128d, _mm_cvtss_sd);
+	Ret_M128d(__m128i, _mm_cvttpd_epi32);
+	Ret_M128(__m128i, _mm_cvttps_epi32);
+	Ret_M128d(int, _mm_cvttsd_si32);
+	Ret_M128d(int64_t, _mm_cvttsd_si64); // _mm_cvttsd_si64x is an alias to this.
+}
+void test_elementarymath()
+{
+	// SSE2 Elementary Math Functions instructions:
+	Ret_M128d(__m128d, _mm_sqrt_pd);
+	Ret_M128d_M128d(__m128d, _mm_sqrt_sd);
+}
+
+void test_generalsupport()
+{
+	// SSE2 General Support instructions:
+	_mm_clflush(interesting_floats);
+	_mm_lfence();
+	_mm_mfence();
+	_mm_pause();
+}
+
+void test_load()
+{
+	// SSE2 Load functions:
+	Ret_DoublePtr(__m128d, _mm_load_pd, 2, 2);
+	Ret_DoublePtr(__m128d, _mm_load_pd1, 1, 1);
+	Ret_DoublePtr(__m128d, _mm_load_sd, 1, 1);
+	Ret_IntPtr(__m128i, _mm_load_si128, __m128i*, 4, 4);
+	Ret_DoublePtr(__m128d, _mm_load1_pd, 1, 1);
+	Ret_M128d_DoublePtr(__m128d, _mm_loadh_pd, double*, 1, 1);
+	Ret_IntPtr(__m128i, _mm_loadl_epi64, __m128i*, 2, 1);
+	Ret_M128d_DoublePtr(__m128d, _mm_loadl_pd, double*, 1, 1);
+	Ret_DoublePtr(__m128d, _mm_loadr_pd, 2, 2);
+	Ret_DoublePtr(__m128d, _mm_loadu_pd, 2, 1);
+	Ret_IntPtr(__m128i, _mm_loadu_si128, __m128i*, 4, 1);
+	Ret_IntPtr(__m128i, _mm_loadu_si32, __m128i*, 1, 1);
+}
+
+void test_logical()
+{
+	// SSE2 Logical instructions:
+	Ret_M128d_M128d(__m128d, _mm_and_pd);
+	M128i_M128i_M128i(_mm_and_si128);
+	Ret_M128d_M128d(__m128d, _mm_andnot_pd);
+	M128i_M128i_M128i(_mm_andnot_si128);
+	Ret_M128d_M128d(__m128d, _mm_or_pd);
+	M128i_M128i_M128i(_mm_or_si128);
+	Ret_M128d_M128d(__m128d, _mm_xor_pd);
+	M128i_M128i_M128i(_mm_xor_si128);
+}
+
+void test_misc()
+{
+	// SSE2 Miscellaneous instructions:
+	Ret_M128i(int, _mm_movemask_epi8);
+	Ret_M128d(int, _mm_movemask_pd);
+	M128i_M128i_M128i(_mm_packs_epi16);
+	M128i_M128i_M128i(_mm_packs_epi32);
+	M128i_M128i_M128i(_mm_packus_epi16);
+}
+
+void test_move()
+{
+	// SSE2 Move instructions:
+	Ret_M128i(__m128i, _mm_move_epi64);
+	Ret_M128d_M128d(__m128d, _mm_move_sd);
+}
+
+void test_probability()
+{
+	// SSE2 Probability/Statistics instructions:
+	M128i_M128i_M128i(_mm_avg_epu16);
+	M128i_M128i_M128i(_mm_avg_epu8);
+}
+
+void test_set()
+{
+/*
+	// SSE2 Set functions:
+	_mm_set_epi16
+	_mm_set_epi32
+	_mm_set_epi64
+	_mm_set_epi64x
+	_mm_set_epi8
+	_mm_set_pd
+	_mm_set_pd1
+	_mm_set_sd
+	_mm_set1_epi16
+	_mm_set1_epi32
+	_mm_set1_epi64
+	_mm_set1_epi64x
+	_mm_set1_epi8
+	_mm_set1_pd
+	_mm_setr_epi16
+	_mm_setr_epi32
+	_mm_setr_epi64
+	_mm_setr_epi8
+	_mm_setr_pd
+	_mm_setzero_pd
+	_mm_setzero_si128
+*/
```

Comment:
What's the status of these?

Reply:
I'll add these as a followup, not yet done. Updated the comment to have a clear "TODO" for easy grepping.